src/docker_client.rs

```diff
@@ -0,0 +1,43 @@
+use std::process::{Command, Stdio};
+
+pub struct DockerClient {}
+
+impl DockerClient {
+    pub fn build_image(ruby_version: &str, rails_version: &str) -> Command {
+        let mut command = Command::new("docker");
+
+        command
+            .arg("build")
+            .arg("--build-arg")
+            .arg(format!("RUBY_VERSION={}", ruby_version))
+            .arg("--build-arg")
+            .arg(format!("RAILS_VERSION={}", rails_version))
+            .arg("-t")
+            .arg(format!("rails-new-{}-{}", ruby_version, rails_version))
+            .arg("-")
+            .stdin(Stdio::piped());
+
+        command
+    }
+
+    pub fn run_image(ruby_version: &str, rails_version: &str, args: Vec<String>) -> Command {
+        let binding = std::env::current_dir().unwrap();
+        let current_dir = binding.to_str().unwrap();
+
+        let mut command = Command::new("docker");
+
+        command
```

Comment:
There's an edge case here I think you might be susceptible to. It's possible this process could fill it's STDOUT buffer and deadlock until something consumes it. Same with the command above

Reply:
That is exactly the kind of review I was hopping for, thanks. Given I never wrote Rust code before, how would I avoid this problem?

Reply:
If I got it correctly, I think https://github.com/rust-lang/rust/commit/ce2d95cd75adb5f75921536191b4dcfa41be1eff solves this problem.

Reply:
Yeah! That should totally work. This is one of my favorite bugs because it's so silly but if it bites you it's a real jerk to debug.